File: atcoder/ABS/cpp/f.cpp
#include<iostream>
using namespace std;

int f(int m){
  int sum=0;
  while(m){
    sum+=m%10;
    m/=10;
  }
  return sum;
}

int main(){
  int n,a,b;
  cin>>n>>a>>b;

  int ans=0;
  for(int i=1;i<=n;i++){
    int c=f(i);
    if(a<=c&&c<=b)
      ans+=i;
  }

  cout<<ans<<endl;
}

File: atcoder/ABS/php/f.php
<?php

function f($d){
  $sum=0;
  while($d>0){
    $sum+=$d%10;
    $d/=10;
  }
  return $sum;
}

list($n,$a,$b)=fscanf(STDIN,"%d %d %d");

$ans=0;

for($i=1;$i<=$n;$i++){
  if($a<=f($i)&&f($i)<=$b)
    $ans+=$i;
}

print $ans;

?>

